src/python/pants/engine/internals/build_files.py

```diff
@@ -112,22 +113,48 @@ class AddressFamilyDir(EngineAwareParameter):
     """
 
     path: str
+    build_files_required: bool = field(default=True, hash=False, compare=False)
 
     def debug_hint(self) -> str:
         return self.path
 
 
+@dataclass(frozen=True)
+class AddressFamilyRequest:
+    defaults: BuildFileDefaults
+    directory: AddressFamilyDir
+
+
+@rule
+async def get_address_family_request(directory: AddressFamilyDir) -> AddressFamilyRequest:
```

Comment:
Is the engine smart enough to know that if any of my `Get`s in a rule changes it needs to re-run as well, so it is not only the input params?

In other words, I thought that I would need to provide the current defaults to the `parse_address_family` rule in the input params for the caching to work properly, rather than relying on `Get`ing the defaults in the rule body, to invalidate properly in case the current defaults change due to changes in a parent BUILD file.

Reply:
Well, heh, after some exploring and thinking I realize that this must be the case, or pretty much anything would not work as it does if it didn't do infer this.. and my experiments shows it does work, so making the change :D